### PR TITLE
fixes to header component

### PR DESCRIPTION
**Pull request by savindi7**
### Related Issues
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- [Related PR `#2877`](https://github.com/wso2/identity-apps/pull/2877)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: modules/react-components/src/components/header/header.tsx [hunk 1]
```diff
@@ -386,18 +386,18 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
         }
 
         const adjustedUserDropdownLinks : HeaderLinkCategoryInterface[] = userDropdownLinks
-            .reduce((previousValue, currentValue) => {
-                const { category, categoryLabel, links } : Partial<HeaderLinkCategoryInterface> = currentValue;
-                const findObj : Partial<HeaderLinkCategoryInterface> = previousValue
+            .reduce((previous, current) => {
```
Comment (brionmario) [inline modules/react-components/src/components/header/header.tsx:389]:
```suggestion
            .reduce((previous: HeaderLinkCategoryInterface, current: HeaderLinkCategoryInterface) => {
```

File: modules/react-components/src/components/header/header.tsx [hunk 2]
```diff
@@ -417,6 +417,10 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
                     {
                         category.links.map((link: HeaderLinkInterface, linkIndex: number) => {
 
+                            if(!link){
```
Comment (brionmario) [inline modules/react-components/src/components/header/header.tsx:420]:
```suggestion
                            if (!link){
```

File: modules/react-components/src/components/header/header.tsx (hunk 1 above)
Comment (savindi7) [reply modules/react-components/src/components/header/header.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2893/commits/98fa9be083704adaa04170de36701b8a87049f93

File: modules/react-components/src/components/header/header.tsx [hunk 3]
```diff
@@ -417,6 +417,10 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
                     {
                         category.links.map((link: HeaderLinkInterface, linkIndex: number) => {
 
+                            if (!link){
```
Comment (brionmario) [inline modules/react-components/src/components/header/header.tsx:420]:
```suggestion
                            if (!link) {
```

File: modules/react-components/src/components/header/header.tsx (hunk 2 above)
Comment (savindi7) [reply modules/react-components/src/components/header/header.tsx]: fixed with https://github.com/wso2/identity-apps/pull/2893/commits/4a061c29fb3ff30eb820adbd861d1a64b839efe1

**Review by brionmario — APPROVED** [review]

File: modules/react-components/src/components/header/header.tsx (hunk 3 above)
Comment (UdeshAthukorala) [inline modules/react-components/src/components/header/header.tsx:420]: Check formatting. Missing space between `)` & `{`.

File: apps/console/src/extensions/configs/models/common.ts [hunk 4]
```diff
@@ -41,9 +41,12 @@ export interface CommonConfig {
         getHeaderSubPanelExtensions: () => HeaderSubPanelItemInterface[];
         /**
          * Get the user dropdown link extensions.
+         * @param {string} tenantDomain - Current tenant
```
Comment (UdeshAthukorala) [inline apps/console/src/extensions/configs/models/common.ts:44]: Missing new line here between method description and parameter descriptions.
Shall we add a `.` at the end?

File: modules/react-components/src/components/header/header.tsx (hunk 3 above)
Comment (savindi7) [reply modules/react-components/src/components/header/header.tsx:420]: Fixed with https://github.com/wso2/identity-apps/pull/3122/commits/6924afc4f3666ebe765dacebb9bb90f418354c61

File: apps/console/src/extensions/configs/models/common.ts (hunk 4 above)
Comment (savindi7) [reply apps/console/src/extensions/configs/models/common.ts:44]: Fixed with https://github.com/wso2/identity-apps/pull/3122/commits/6924afc4f3666ebe765dacebb9bb90f418354c61

**Pull request merged**